### ISSUE TITLE
Undefined array key "vjs_sync" and other issues

### ISSUE THREAD

**Issue by brentkearney** [Piwigo/piwigo-videojs#200]
Plugin version: 2.9b
Piwigo version: 13.6.0
Theme: Bootstrap Darkroom Version 2.5.17

I installed Piwigo using the [linuxserver docker image](https://hub.docker.com/r/linuxserver/piwigo). It mostly works, but video posters are not being generated, so I'm looking deeper into it, and here are some issues I have found.

In Piwigo Admin -> Plugins -> VideoJS Settings, at the bottom of the page, I see:
```
 Warning: Undefined array key "resolution" in /config/www/_data/templates_c/p6jhns^b6abb1d84350daf359dafd4908d870913a76b342_0.file.admin_config.tpl.php on line 268
/> Yes   Warning: Undefined array key "resolution" in /config/www/_data/templates_c/p6jhns^b6abb1d84350daf359dafd4908d870913a76b342_0.file.admin_config.tpl.php on line 270
checked="checked"/> No 
Resolution switcher.
```

When I click the `Synchronize` tab, everything under **Program Dependencies** has a green checkmark next to it. I also connected to the running container, and verified that all of the binaries exist (all in `/usr/bin`), and they all run. I can manually create video posters with `ffmpeg`.

I have these selections in the form:
- [x] Create a poster at position in second:   4
- [x] Overwrite existing posters 
- [x] Output format: JPG
- [ ] perform a simulation only (nothing will be changed in the database)
- [x] Search in sub-albums

When I click the Submit button, I get this output:
```
Warning: Undefined array key "vjs_sync" in /config/www/plugins/piwigo-videojs/admin/admin_sync.php on line 78

Fatal error: Uncaught TypeError: array_merge(): Argument #1 must be of type array, bool given in /config/www/plugins/piwigo-videojs/admin/admin_sync.php:78 Stack trace: #0 /config/www/plugins/piwigo-videojs/admin/admin_sync.php(78): array_merge() #1 /config/www/plugins/piwigo-videojs/admin/admin.php(68): include_once('...') #2 /app/www/public/admin/plugin.php(53): include_once('...') #3 /app/www/public/admin.php(345): include('...') #4 {main} thrown in /config/www/plugins/piwigo-videojs/admin/admin_sync.php on line 78
```

Searching the issues, I found this [old one from 2018](https://github.com/Piwigo/piwigo-videojs/issues/130). Has there been a regression?

In a log file in `config/www/_data/logs/`, I discovered what looks like a missing `/` in the paths to the binaries:

```
[2023-03-05 18:17:31]	[INFO]	upload_file_video, $file_path = ././upload/2023/03/06/20230306011731-9436fdc2.mp4, $representative_ext = 
[2023-03-05 18:17:31]	[INFO]	upload_file_video, Poster at 2s
[2023-03-05 18:17:31]	[DEBUG]	upload_file_video, Tried /usr/binffmpeg -ss 2 -i "././upload/2023/03/06/20230306011731-9436fdc2.mp4" -frames:v 1 "././upload/2023/03/06/pwg_representative/20230306011731-9436fdc2.jpg"
[2023-03-05 18:17:31]	[DEBUG]	sh: /usr/binffmpeg: not found
[2023-03-05 18:17:31]	[DEBUG]	upload_file_video, Tried /usr/binavconv -ss 2 -i "././upload/2023/03/06/20230306011731-9436fdc2.mp4" -frames:v 1 "././upload/2023/03/06/pwg_representative/20230306011731-9436fdc2.jpg"
[2023-03-05 18:17:31]	[DEBUG]	sh: /usr/binavconv: not found
[2023-03-05 18:17:31]	[INFO]	Handling ././upload/2023/03/06/20230306011731-9436fdc2.mp4 got 
[2023-03-05 18:17:31]	[INFO]	add_uploaded_file : force cache generation, derivative_url = https://media.heartburn.love/i.php?/upload/2023/03/06/20230306011731-9436fdc2-me.mp4
[2023-03-05 18:17:32]	[DEBUG]	community_ws_images_uploadCompleted : enter notification part
[2023-03-05 18:17:32]	[DEBUG]	empty_lounge, exec=2pDd, begins
[2023-03-05 18:17:32]	[DEBUG]	empty_lounge, exec=2pDd wins the race and gets the token!
[2023-03-05 18:17:32]	[DEBUG]	empty_lounge, exec=2pDd, ends
[2023-03-05 18:18:40]	[ERROR]	[i.php]	400 Invalid chars in request
  url: '/i.php?/upload/2023/03/06/20230306011731-9436fdc2-th.mp4?1678065520'
[2023-03-05 18:18:43]	[ERROR]	[i.php]	400 Invalid chars in request
  url: '/i.php?/upload/2023/03/06/20230306011731-9436fdc2-th.mp4?1678065523'
[2023-03-05 18:45:51]	[ERROR]	[i.php]	400 Invalid chars in request
  url: '/i.php?/upload/2023/03/06/20230306011731-9436fdc2-th.mp4?1678067151'
[2023-03-05 18:45:54]	[ERROR]	[i.php]	400 Invalid chars in request
  url: '/i.php?/upload/2023/03/06/20230306011731-9436fdc2-th.mp4?1678067153'
[2023-03-05 18:46:37]	[ERROR]	[i.php]	400 Invalid chars in request
  url: '/i.php?/upload/2023/03/06/20230306011731-9436fdc2-th.mp4?1678067197'
```

I'm not sure what the "400 Invalid chars in request" refers to, but in the Edit Photo section of the Admin area for a video, where there should be a thumbnail/poster, there is a "missing image" blue question mark icon.

The biggest problem for my immediate concern (working video posters) is probably the broken bin paths, so I will see what hackery I can do to fix that. I am open to any suggestions - I haven't used PHP in awhile but I'm fairly familiar with it.

**Comment by brentkearney** [comment]
The binary path problem was because I had set `$conf['ffmpeg_dir'] = '/usr/bin';` in `./www/local/config/config.inc.php` during my initial configuration. After changing it to '/usr/bin/', the poster image files are created upon video upload. However, they still don't appear in the web browser.

I also fixed the plugin form submission, which generated all of the poster files for the uploaded videos - but they still don't appear in the web browser. That was fixed by adding a conditional in `admin_sync.php:78`:
```
    // Merge default value with user configuration
    if (isset($conf['vjs_sync']))
    {
        $sync_options = array_merge(unserialize($conf['vjs_sync']), $sync_options_form);
    }
    else
    {
        $sync_options = array_merge($sync_options, $sync_options_form);
    }
```

I pulled the code from the master branch so that I could submit a PR, only to discover that the files were significantly different than what was installed via the Piwigo interface. The form works with the master branch code, but alas, the the poster images still do not appear.

**Comment by brentkearney** [comment]
> I'm not sure what the "400 Invalid chars in request" refers to

That referred to the local config option for _Permitted characters_, `$conf['sync_chars_regex'] =`. I had it set for the first suggested regex, and then tried the more permissive one for all Unicode characters. This reduced the `ERROR` entries in the log a lot, but there will still a few. I just commented out that option, and they went away. Hopefully it's not too crazy insecure now.

I'm going to close this issue because most of what is mentioned were configuration problems that have been solved. The worst problem -- the poster images are not displayed, the error png from the default template is displayed instead -- remains unsolved, but should be it's own issue.
